tests/simulator_test.cpp

```diff
@@ -1,2 +1,50 @@
+#include "constants.h"
+#include "main/simulator.h"
 #include "gtest/gtest.h"
 
+using constants::kDefaultCollectorPopulation;
+using constants::kDefaultDepth;
+using constants::kDefaultEnergyCollectorRadius;
+using constants::kDefaultNumRaysPerRow;
+using constants::kDefaultSimulationRadius;
+using constants::kDefaultSourcePower;
+using constants::kSkipFreq;
+using constants::kSkipPath;
+using core::Ray;
+using core::RayHitData;
+using core::Vec3;
+
+const Vec3 kDefaultSourcePosition(0, 0, constants::kDefaultSimulationRadius);
+const Vec3 kVecStraightDown(0, 0, -1);
+const float kDefaultEnergyPerRay =
+    constants::kDefaultSourcePower /
+    (constants::kDefaultNumRaysPerRow * constants::kDefaultNumRaysPerRow);
+
+class ReferencePlateTest : public ::testing::Test {
+public:
+  ReferencePlateTest()
+      : sim(Simulator(kSkipFreq, kDefaultNumRaysPerRow,
+                      kDefaultCollectorPopulation, kDefaultSimulationRadius,
+                      kSkipPath, kDefaultDepth, kDefaultSourcePower)){};
+
+protected:
+  Simulator sim;
+};
+
+TEST_F(ReferencePlateTest, EnergyCollectionFromRay) {
+
+  RayHitData hitData;
+
+  Ray starightDown(kDefaultSourcePosition, kVecStraightDown,
+                   kDefaultEnergyPerRay);
+  sim.rayTrace(starightDown, &hitData, 2);
```

Comment:
What 2 means? A good way to documenting call sites is calling out argument names explicitly e.g:

```c++
sim.rayTrace(starightDown, &hitData, /*depth=*/2);
```

clang also warns you in case you get the name wrong, so that there is some additional check that you are passing the right param.